Maths/Fraction.js

```diff
@@ -0,0 +1,67 @@
+/**
+ * @function fraction
+ * @description This function returns the fraction of a float type number.
+ * @param {number} num - a float type number is expected, but an integer will also work.
+ * @param {number} accuracy - the accuracy of the fraction, the default is 6. like if the accuracy is 2 then for 1.333 result will be 10/9, where for 6 it will return 1333/1000.
+ * @return {Array} - an array containing the numerator and denominator of the fraction.
+ * @see https://en.wikipedia.org/wiki/Repeating_decimal and 
+ * @see https://en.wikipedia.org/wiki/Decimal#Decimal_fractions
+ * @example fraction(0.5) // [1, 2]
+ * @example fraction(0.3333333333333333) // [1, 3]
+ * @example fraction(0.25) // [1, 4]
+ * @example fraction(5.56) // [139, 25]
+ * @example fraction(0.33) // [33, 100]
+ * @example fraction(0.33,2) // [10, 3] 
+ */
+function fraction(number, accuracy = 6) {
+    if (typeof number === "number" && !Number.isNaN(number) && Number.isFinite(number)
+        && typeof accuracy === "number" && !Number.isNaN(accuracy) && accuracy >= 1 && accuracy <= 16) {
+        let neg = 1;
+        // if number is a negative then following code will run
+        if (number < 0) {
+            neg = -1;
+            number = Math.abs(number);
+        }
+        // if number is a 0 then it will return [0, 1]
+        if (number === 0) return [0, 1];
+        if (Number.isInteger(number)) return [neg * number, 1];
+        // if number is a not an integer then following code will run
+        number = number.toString();
+        let len;
+        let reg = number.match(/(\d+?)\1+$/);
```

Comment:
This is a hacky way to deal with "repeating" fractional parts, and I'd argue that it solves the wrong problem. Since we're dealing with finite-size numbers (floats) here, there *are* no "repeating" decimals; we can't distinguish a "repeating" decimal from a "deliberately ending decimal". We need to treat both the same (it seems "accuracy" is more of a "threshold" of when you consider a decimal repeating?).

In fact, all floats can even be represented *exactly* as fractions (though perhaps maybe not as fractions of floats, that might get hairy towards some edge cases) where the denominator is a power of two and the numerator is an integer.

I wouldn't exclude the possibility that some special provisions for repeating decimals may be useful, but then what problem this exactly solves (and how) and what guarantees it makes should be explicitly stated.